plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/manifest/ManifestManager.kt

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.editor.context.project.manifest
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.util.io.HttpRequests
+import com.intellij.util.system.CpuArch
+import software.aws.toolkits.core.utils.getLogger
+import java.net.HttpURLConnection
+import java.net.URL
+
+class ManifestManager {
+    // TODO: switch to prod url
+    private val cloudFrontUrl = "https://aws-toolkit-language-servers.amazonaws.com/temp/manifest.json"
+    val SERVER_VERSION = "0.1.0"
+    val currentOs = getOs()
+    private val arch = CpuArch.CURRENT
+    private val mapper = ObjectMapper()
```

Comment:
should use `jacksonObjectMapper()` as it's for Kotlin. 
otherwise we need to do `ObjectMapper().registerModule(KotlinModule())`